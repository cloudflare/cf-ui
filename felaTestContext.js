// this is a wrapper for testing, it passes fela based
// components fela's renderer and global theme

import React from 'react';
import { Provider, ThemeProvider } from 'react-fela';
import { createRenderer } from './felaRenderer';
import { variables } from 'cf-style-const';

const renderer = createRenderer({ useMonolithic: true });

export default component => (
  <Provider renderer={renderer}>
    <ThemeProvider theme={variables}>{component}</ThemeProvider>
  </Provider>
);
