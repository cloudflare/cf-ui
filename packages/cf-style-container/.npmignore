node_modules
*.log
src
coverage
test
example
karma.conf.js
