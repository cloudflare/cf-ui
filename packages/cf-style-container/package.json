{
  "name": "cf-style-container",
  "description": "Cloudflare Style Container",
  "version": "0.0.2",
  "main": "lib/index.js",
  "author": "Vojtech Miksu <vojtech@cloudflare.com>",
  "license": "BSD-3-Clause",
  "publishConfig": {
    "registry": "http://registry.npmjs.org/"
  },
  "dependencies": {
    "react-fela": "^4.2.5"
  },
  "peerDependencies": {
    "react": "^0.14.0 || ^15.0.0-0 || ^16.0.0-0"
  }
}
