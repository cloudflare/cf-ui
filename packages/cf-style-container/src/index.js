import React, { Component, PropTypes } from 'react';
import { combineRules } from 'fela';
import {
  createComponent as createFelaComponent,
  ThemeProvider,
  connect
} from 'react-fela';

const createComponent = (rule, type = 'div', passThroughProps = []) =>
  createFelaComponent(
    rule,
    type,
    type.propTypes
      ? passThroughProps.concat(Object.keys(type.propTypes))
      : passThroughProps
  );

const applyTheme = (ComponentToWrap, theme) => {
  class ThemedComponent extends Component {
    getChildContext() {
      return {
        theme: {
          ...(this.context.theme || {}),
          ...theme(this.context.theme)
        }
      };
    }

    render() {
      return <ComponentToWrap {...this.props} />;
    }
  }

  ThemedComponent.childContextTypes = { theme: PropTypes.object };
  ThemedComponent.contextTypes = { theme: PropTypes.object };
  return ThemedComponent;
};

const createComponentStyles = (styleFunctions, component) => {
  const mapStylesToProps = props =>
    renderer => {
      const toRender = {};
      for (const style in styleFunctions) {
        toRender[style] = renderer.renderRule(styleFunctions[style], props);
      }
      return toRender;
    };
  return connect(mapStylesToProps)(component);
};

export {
  createComponent,
  applyTheme,
  ThemeProvider,
  connect,
  combineRules,
  createComponentStyles
};
