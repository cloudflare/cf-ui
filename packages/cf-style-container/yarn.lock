# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
# yarn lockfile v1


balanced-match@0.1.0:
  version "0.1.0"
  resolved "https://registry.yarnpkg.com/balanced-match/-/balanced-match-0.1.0.tgz#b504bd05869b39259dd0c5efc35d843176dccc4a"

clone@^1.0.2:
  version "1.0.2"
  resolved "https://registry.yarnpkg.com/clone/-/clone-1.0.2.tgz#260b7a99ebb1edfe247538175f783243cb19d149"

color-convert@^1.3.0:
  version "1.9.0"
  resolved "https://registry.yarnpkg.com/color-convert/-/color-convert-1.9.0.tgz#1accf97dd739b983bf994d56fec8f95853641b7a"
  dependencies:
    color-name "^1.1.1"

color-name@^1.0.0, color-name@^1.1.1:
  version "1.1.2"
  resolved "https://registry.yarnpkg.com/color-name/-/color-name-1.1.2.tgz#5c8ab72b64bd2215d617ae9559ebb148475cf98d"

color-string@^0.3.0:
  version "0.3.0"
  resolved "https://registry.yarnpkg.com/color-string/-/color-string-0.3.0.tgz#27d46fb67025c5c2fa25993bfbf579e47841b991"
  dependencies:
    color-name "^1.0.0"

color@^0.11.0:
  version "0.11.4"
  resolved "https://registry.yarnpkg.com/color/-/color-0.11.4.tgz#6d7b5c74fb65e841cd48792ad1ed5e07b904d764"
  dependencies:
    clone "^1.0.2"
    color-convert "^1.3.0"
    color-string "^0.3.0"

css-color-function@^1.3.0:
  version "1.3.0"
  resolved "https://registry.yarnpkg.com/css-color-function/-/css-color-function-1.3.0.tgz#72c767baf978f01b8a8a94f42f17ba5d22a776fc"
  dependencies:
    balanced-match "0.1.0"
    color "^0.11.0"
    debug "~0.7.4"
    rgb "~0.1.0"

css-in-js-utils@^1.0.3:
  version "1.0.3"
  resolved "https://registry.yarnpkg.com/css-in-js-utils/-/css-in-js-utils-1.0.3.tgz#9ac7e02f763cf85d94017666565ed68a5b5f3215"
  dependencies:
    hyphenate-style-name "^1.0.2"

debug@~0.7.4:
  version "0.7.4"
  resolved "https://registry.yarnpkg.com/debug/-/debug-0.7.4.tgz#06e1ea8082c2cb14e39806e22e2f6f757f92af39"

fela@^4.3.2:
  version "4.3.2"
  resolved "https://registry.yarnpkg.com/fela/-/fela-4.3.2.tgz#1e407351bf26d618665bc1bc19708cfbce78bd03"
  dependencies:
    css-in-js-utils "^1.0.3"

hyphenate-style-name@^1.0.2:
  version "1.0.2"
  resolved "https://registry.yarnpkg.com/hyphenate-style-name/-/hyphenate-style-name-1.0.2.tgz#31160a36930adaf1fc04c6074f7eb41465d4ec4b"

react-fela@^4.2.5:
  version "4.3.2"
  resolved "https://registry.yarnpkg.com/react-fela/-/react-fela-4.3.2.tgz#12703afceacca43fb8dcb093917ba6793a5d149e"

rgb@~0.1.0:
  version "0.1.0"
  resolved "https://registry.yarnpkg.com/rgb/-/rgb-0.1.0.tgz#be27b291e8feffeac1bd99729721bfa40fc037b5"
