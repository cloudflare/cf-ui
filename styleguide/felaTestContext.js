import React from 'react';
import { Provider } from 'react-fela';
import { createRenderer } from './felaRenderer';

const renderer = createRenderer({ useMonolithic: true });

export default component => (
  <Provider renderer={renderer}>{component}</Provider>
);
