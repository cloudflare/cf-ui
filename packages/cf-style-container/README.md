# cf-style-container

> Cloudflare Style Container

Set of high order components and other helpers for fela based applications.

## Installation

```sh
$ npm install cf-style-container
```

### Aliased functions from fela and react-fela

We proxy/alias some useful functions from fela without changing their behaviour. See the original documentation for more details. We wrap all Fela APIs so we can eventually switch Fela to a different CSS in JS lib if ever needed.

- [combineRules](https://github.com/rofrischmann/fela/blob/master/docs/api/fela/combineRules.md)
- [connect](https://github.com/rofrischmann/fela/blob/master/packages/react-fela/docs/connect.md)
- [ThemeProvider](https://github.com/rofrischmann/fela/blob/master/packages/react-fela/docs/ThemeProvider.md)

### createComponent(rule, [type])

Very similar to [createComponent](https://github.com/rofrischmann/fela/blob/master/packages/react-fela/docs/createComponent.md) from react-fela. However, it automatically adds PropTypes from `[type]` in case that it is a React Component.

You should use this HOC every time when you want to use Fela in your component and you need only one className (one rule function).

```jsx
import React, { PropTypes } from 'react';
import { createComponent } from 'cf-style-container';

const styles = ({ theme, size }) => ({
  fontWeight: theme[`fontWeight${size}`],
  fontSize: theme[`fontSize${size}`],
  lineHeight: theme[`lineHeight${size}`],
  marginTop: theme[`marginTop${size}`]
});

const Heading = ({ size, className, children }) => {
  const tagName = 'h' + size;
  return React.createElement(tagName, { className }, children);
};

Heading.propTypes = {
  size: PropTypes.oneOf([1, 2, 3, 4, 5, 6]).isRequired,
  className: PropTypes.string.isRequired,
  children: PropTypes.node
};

export default createComponent(styles, Heading);
```

### createComponentStyles(rules, Component)

Useful when you need multiple classNames (and rules) in one component.

```jsx
import React, { PropTypes } from 'react';
import { createComponentStyles } from 'cf-style-container';

const mainStyles = ({ theme }) => ({
  margin: theme.main.margin,
  padding: theme.main.padding,
});

const legendStyles = ({ theme }) => ({
  padding: theme.legend.padding,
  marginBottom: theme.legend.marginBottom,
  borderBottom: theme.legend.borderBottom,
});

const FormFieldset = ({ legend, styles }) => (
  <fieldset className={styles.mainStyles}>
    {' '}
    <legend className={styles.legendStyles}>
      {legend}
    </legend>
  </fieldset>
);

FormFieldset.propTypes = {
  styles: PropTypes.object.isRequired,
  legend: PropTypes.string.isRequired
};

export default createComponentStyles({ mainStyles, legendStyles }, FormFieldset);
```

Notice that rules are now an object. The names you chose will be used for classNames
accessible as `styles.mainStyles` and `styles.legendStyles` in this case.

### applyTheme(Component, theme)

And HOC that ties a Fela component with the theme (adds the theme to its context).

```jsx
import HeadingUnstyled from './Heading';
import HeadingTheme from './HeadingTheme';

import { applyTheme } from 'cf-style-container';

const Heading = applyTheme(HeadingUnstyled, HeadingTheme);

// themed component
<Heading />
```

