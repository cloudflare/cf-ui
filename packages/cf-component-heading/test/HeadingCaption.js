import React from 'react';
import {
  HeadingCaption,
  HeadingCaptionTheme
} from '../../cf-component-heading/src/index';
import renderer from 'react-test-renderer';
import felaTestContext from '../../../felaTestContext';
import { applyTheme } from 'cf-style-container';

const StyledHeadingCaption = applyTheme(HeadingCaption, HeadingCaptionTheme);

test('should render', () => {
  const component = renderer.create(
    felaTestContext(
      <StyledHeadingCaption>Heading Caption</StyledHeadingCaption>
    )
  );
  expect(component.toJSON()).toMatchSnapshot();
});
