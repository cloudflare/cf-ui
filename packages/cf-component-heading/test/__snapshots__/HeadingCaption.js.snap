// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
<small
  className="cf-1d38fhm"
>
  Heading Caption
</small>
`;

exports[`should render 2`] = `
".cf-1d38fhm {
  color: #7c7c7c;
  font-color: #333;
  font-weight: 400;
  margin-left: .5em
}"
`;
